test/OpenTelemetry.Instrumentation.GrpcClient.Tests/OpenTelemetry.Instrumentation.GrpcClient.Tests.csproj

```
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>Unit test project for OpenTelemetry GrpcClient instrumentation</Description>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Protobuf Include="Proto\greet.proto" GrpcServices="Client;Server" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
+    <PackageReference Include="Moq" Version="4.11.0" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+    </PackageReference>
+    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
+
+    <PackageReference Include="Google.Protobuf" Version="3.12.3" />
+    <PackageReference Include="Grpc.AspNetCore" Version="2.30.0-pre1" />
+    <PackageReference Include="Grpc.Net.Client" Version="2.30.0-pre1" />
+    <PackageReference Include="Grpc.Tools" Version="2.30.0-pre1" PrivateAssets="All" />
```

Will fix build error:

```suggestion
    <PackageReference Include="Grpc.Tools" Version="2.25.0" PrivateAssets="All" />
```

---

Ah, ok. The reason I didn't do this at first was because doing so renders the warning:

```
Error | NU1605 | Detected package downgrade: Grpc.Tools from 2.30.0-pre1 to 2.25.0. Reference the package directly from the project to select a different version.  OpenTelemetry.Instrumentation.GrpcClient.Tests -> Grpc.AspNetCore 2.30.0-pre1 -> Grpc.Tools (>= 2.30.0-pre1)  OpenTelemetry.Instrumentation.GrpcClient.Tests -> Grpc.Tools (>= 2.25.0)
```

For now also added `<NoWarn>$(NoWarn);NU1605</NoWarn>` to the project.

---

You should be able to avoid the warning and NoWarn by changing `Grpc.AspNetCore` package reference to `Grpc.AspNetCore.Server`.

---

👌 yep, that did it. Thanks for your help!